clippy_lints/src/methods/mod.rs

```diff
@@ -889,6 +889,23 @@ declare_clippy_lint! {
     "using `.into_iter()` on a reference"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Checks for calls to `map` followed by a `count`.
+    ///
+    /// **Why is this bad?** It looks suspicious. Maybe `map` was confused with `filter`.
+    ///
+    /// **Known problems:** None
+    ///
+    /// **Example:**
+    ///
+    /// ```rust
+    /// let _ = (0..3).map(|x| x + 2).count();
+    /// ```
+    pub SUSPICIOUS_MAP,
+    pedantic,
```

Comment:
I think this could even be a `style` (Using map for side effects should be rewritten) or `complexity` (Using map for side effects might be confusing and can be written less complex) lint.

Reply:
You're right. I just thought it was too invasive at first

Reply:
Let's try it. If people complain, we can still move it to pedantic.